File: src/main/java/dev/slne/spawn/trader/manager/object/impl/LightTrade.java
package dev.slne.spawn.trader.manager.object.impl;

import dev.slne.spawn.trader.manager.object.Trade;
import dev.slne.spawn.trader.util.ItemBuilder;
import java.util.List;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

/**
 * The type Light trade.
 */
public class LightTrade implements Trade {

  @Override
  public List<ItemStack> requirements() {
    return List.of(new ItemBuilder(Material.EMERALD, 5).build(),
        new ItemBuilder(Material.REDSTONE_LAMP, 20).build());
  }

  @Override
  public List<ItemStack> rewards() {
    return List.of(new ItemBuilder(Material.LIGHT, 20).build());
  }

  @Override
  public Integer id() {
    return 1;
  }

  @Override
  public String rewardMessage() {
    return "Danke für den Einkauf von 20x Licht-Blöcken!";
  }
}

File: src/main/java/dev/slne/spawn/trader/manager/trade/impl/LightTrade.kt
package dev.slne.spawn.trader.manager.trade.impl

import dev.slne.spawn.trader.manager.trade.Trade
import dev.slne.surf.surfapi.bukkit.api.builder.buildItem
import dev.slne.surf.surfapi.core.api.util.mutableObjectListOf
import org.bukkit.Material

class LightTrade : Trade {
    override fun requirements() = mutableObjectListOf(
        buildItem(Material.EMERALD, 5) {},
        buildItem(Material.REDSTONE_LAMP, 20) {}
    )

    override fun rewards() = mutableObjectListOf(buildItem(Material.LIGHT, 20) {})

    override fun id(): Int {
        return 1
    }

    override fun name(): String {
        return "light-block"
    }

    override fun rewardMessage(): String {
        return "Danke für den Einkauf von 20x Licht-Blöcken!"
    }

    override fun cooldown(): Long {
        return 75600000L
    }
}
